### ISSUE TITLE
Empty cookie name crash CookieCollection

### ISSUE THREAD

**Issue by Erwane** [cakephp/cakephp#12311]
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.6.6

### What you did
Http\Client::get('http://zephyr-os.fr/');

### What happened
Not an InvalidArgumentException

### What you expected to happen
This domain set an empty cookie header before redirect : 
```
Cache-Control: max-age=0
Connection: Keep-Alive
Content-Encoding: gzip
Content-Type: text/html; charset=utf-8
Date: Mon, 02 Jul 2018 06:55:47 GMT
Expires: Mon, 02 Jul 2018 06:55:47 GMT
Keep-Alive: timeout=3, max=1000
location: http://zephyr-os.zephyr2.atester.fr/
Server: Apache
Set-Cookie: 
Transfer-Encoding: chunked
Vary: Accept-Encoding
```
The Cookie\CookieCollection::parseSetCookieHeader() should ignore empty lines

**Comment by markstory** [comment]
Pull request up now.

**Comment by Erwane** [comment]
Thanks a lot

**Comment by devorases** [comment]
Any chance this fix could be merged into 3.5.x?

**Comment by markstory** [comment]
@devorases Not likely. 3.5 is only getting security fixes right now.
